### PR TITLE
Switch to GitHub Actions

### PR DESCRIPTION
**Pull request by zhen-zen**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+      - run: xcodebuild analyze -quiet -scheme Package -target Package -configuration Debug CLANG_ANALYZER_OUTPUT=plist-html CLANG_ANALYZER_OUTPUT_DIR="$(pwd)/clang-analyze" && [ "$(find clang-analyze -name "*.html")" = "" ]
+      - run: xcodebuild clean -quiet -scheme Package
+      - run: xcodebuild analyze -quiet -scheme Package -target Package -configuration Release CLANG_ANALYZER_OUTPUT=plist-html CLANG_ANALYZER_OUTPUT_DIR="$(pwd)/clang-analyze" && [ "$(find clang-analyze -name "*.html")" = "" ]
+
+  analyze-coverity:
+    name: Analyze Coverity
+    runs-on: macos-latest
+    env:
+      JOB_TYPE: COVERITY
+    if: github.event_name != 'pull_request'
```
Comment (vit9696) [inline .github/workflows/main.yml]: This one needs to become `github.repository == 'acidanthera/OpenCorePkg' && github.event_name != 'pull_request'` as Coverity cannot run in the forks without tokens.

**Comment by vit9696** [comment]
This looks right in time to me, thank you for your help. I mentioned a minor thing with Coverity, which we agreed to change in all the repositories recently. The rest looks good to me. @dhinakg does this look ok to you? If so, I suggest to merge.

File: .github/workflows/main.yml (hunk 1 above)
Comment (dhinakg) [reply .github/workflows/main.yml]:
```suggestion
    if: github.repository_owner == 'acidanthera' && github.event_name != 'pull_request'
```

**Comment by dhinakg** [comment]
Apart from the Coverity job restriction, everything looks fine to me. I have a slight nitpick with some of the ordering - checkout should be first as if we ever decide to download anything into the project folder in the CI bootstrap script, it will be deleted, as checkout deletes everything in the current directory - but I did that mistake in Lilu too, and currently it is not a big deal.

**Pull request merged**